sandwich/src/Test/Sandwich/Interpreters/PruneTree.hs

```diff
@@ -0,0 +1,49 @@
+
+-- | Prunes any nodes and their children in a spec tree that match any of a list of names
+
+module Test.Sandwich.Interpreters.PruneTree (pruneTree) where
+
+import Control.Monad.Free
+import qualified Data.List as L
+import Test.Sandwich.Types.Spec
+
+pruneTree :: Free (SpecCommand context m) () -> [String] -> Free (SpecCommand context m) ()
+pruneTree tree pruneLabels =
+  go tree
+  where
+    go :: Free (SpecCommand context m) () -> Free (SpecCommand context m) ()
+    go = \case
+      (Free (It'' loc no label' ex next))
+        | label' `doesNotMatchAny` pruneLabels -> Free (It'' loc no label' ex (go next))
+        | otherwise -> go next
+      (Free (Introduce'' loc no label' cl alloc cleanup subspec next))
+        | label' `doesNotMatchAny` pruneLabels ->
+          case go subspec of
+            (Pure _) -> go next
+            subspec' -> Free (Introduce'' loc no label' cl alloc cleanup subspec' (go next))
+        | otherwise ->
+          go next
```

Comment:
Nit: the normal style in this project isn't to make a newline for short lines like here and below (and `go tree` above)

Reply:
Done.